crates/storage/src/managers/l1.rs

```diff
@@ -0,0 +1,124 @@
+use std::sync::Arc;
+
+use strata_db::{
+    traits::{BlockStatus, Database, L1Database, L1DatabaseAsync},
+    DbResult,
+};
+use strata_mmr::CompactMmr;
+use strata_primitives::{
+    buf::Buf32,
+    l1::{L1BlockManifest, L1TxRef},
+};
+use strata_state::{block::L2BlockBundle, header::L2Header, id::L2BlockId, l1::L1Tx};
+use threadpool::ThreadPool;
+use tokio::sync::oneshot::{self, error::RecvError};
+use tracing::warn;
+
+use crate::cache::{self, CacheTable};
+
+/// Caching manager of L1 blocks in the block database.
+pub struct L1BlockManager<DB>
```

Comment:
One of the major reasons to have managers was to be free from these Db generics. The `Ops` construct was to facilitate this.

Reply:
You have to pass it down through everything that wants to interact with the database / manager, which becomes cumbersome.